clap_derive/src/derives/args.rs

```diff
@@ -228,7 +228,22 @@ pub fn gen_augment(
                     ParserKind::TryFromOsStr => quote_spanned! { func.span()=>
                         .validator_os(|s| #func(s).map(|_: #convert_type| ()))
                     },
-                    _ => quote!(),
+                    ParserKind::FromStr
+                    | ParserKind::FromOsStr
+                    | ParserKind::FromFlag
+                    | ParserKind::FromOccurrences => quote!(),
```

Comment:
Why not `_`?

Reply:
1. So we take advantage of the compiler so we get notified to re-evaluate this when new variants are added, rather than hoping we catch it in review.